File: Client/logger/categories.ts
/**
 * Logger Category Definitions
 * Type-safe category constants and helpers
 */

export type LogCategory =
  | "POLLING"
  | "PAGES"
  | "HOOKS"
  | "AUTH"
  | "HTTP"
  | "API"
  | "ERRORS"
  | "SECURITY"
  | "SEARCH"
  | "MAP_RENDERING"
  | "NEGOTIATION"
  | "CHECKLISTS"
  | "CALENDAR"
  | "DASHBOARD"
  | "MESSAGES"
  | "FEED";

export type ApiSubcategory =
  | "INITIAL_LOAD"
  | "POLLING"
  | "PAGE_MOUNT"
  | "OTHER";

export const LOG_CATEGORIES = {
  POLLING: "POLLING",
  PAGES: "PAGES",
  HOOKS: "HOOKS",
  AUTH: "AUTH",
  HTTP: "HTTP",
  API: "API",
  ERRORS: "ERRORS",
  SECURITY: "SECURITY",
  SEARCH: "SEARCH",
  MAP_RENDERING: "MAP_RENDERING",
  NEGOTIATION: "NEGOTIATION",
  CHECKLISTS: "CHECKLISTS",
  CALENDAR: "CALENDAR",
  DASHBOARD: "DASHBOARD",
  MESSAGES: "MESSAGES",
  FEED: "FEED",
} as const;

export const API_SUBCATEGORIES = {
  INITIAL_LOAD: "INITIAL_LOAD",
  POLLING: "POLLING",
  PAGE_MOUNT: "PAGE_MOUNT",
  OTHER: "OTHER",
} as const;

/**
 * Map API subcategory to config key
 */
export function apiSubcategoryToConfigKey(subcategory: ApiSubcategory): string {
  const mapping: Record<ApiSubcategory, string> = {
    INITIAL_LOAD: "initialLoad",
    POLLING: "polling",
    PAGE_MOUNT: "pageMount",
    OTHER: "other",
  };
  return mapping[subcategory];
}

/**
 * Map category to config key
 */
export function categoryToConfigKey(category: LogCategory): string {
  const mapping: Record<LogCategory, string> = {
    POLLING: "polling",
    PAGES: "pages",
    HOOKS: "hooks",
    AUTH: "auth",
    HTTP: "http",
    API: "api",
    ERRORS: "errors",
    SECURITY: "security",
    SEARCH: "search",
    MAP_RENDERING: "mapRendering",
    NEGOTIATION: "negotiation",
    CHECKLISTS: "checklists",
    CALENDAR: "calendar",
    DASHBOARD: "dashboard",
    MESSAGES: "messages",
    FEED: "feed",
  };
  return mapping[category];
}

/**
 * Check if a category should always be enabled (errors and security)
 */
export function isAlwaysEnabled(category: LogCategory): boolean {
  return category === "ERRORS" || category === "SECURITY";
}

File: Server/logger/categories.py
"""
Logger Category Definitions
Type-safe category constants and helpers
"""

from enum import Enum

try:
    from enum import StrEnum
except ImportError:
    # StrEnum added in Python 3.11; provide fallback for 3.10 (noqa: intentional)
    class StrEnum(str, Enum):  # noqa: UP042
        """StrEnum fallback for Python < 3.11"""

        pass


class LogCategory(StrEnum):
    """Log category enumeration matching frontend categories"""

    POLLING = "POLLING"
    PAGES = "PAGES"
    HOOKS = "HOOKS"
    AUTH = "AUTH"
    HTTP = "HTTP"
    API = "API"
    ERRORS = "ERRORS"
    SECURITY = "SECURITY"
    POLYGON_SEARCH = "POLYGON_SEARCH"
    DOCUSIGN = "DOCUSIGN"


# Constant dict for easy access
LOG_CATEGORIES: dict[str, LogCategory] = {
    "POLLING": LogCategory.POLLING,
    "PAGES": LogCategory.PAGES,
    "HOOKS": LogCategory.HOOKS,
    "AUTH": LogCategory.AUTH,
    "HTTP": LogCategory.HTTP,
    "API": LogCategory.API,
    "ERRORS": LogCategory.ERRORS,
    "SECURITY": LogCategory.SECURITY,
    "POLYGON_SEARCH": LogCategory.POLYGON_SEARCH,
    "DOCUSIGN": LogCategory.DOCUSIGN,
}


def category_to_config_key(category: LogCategory) -> str:
    """
    Map category to config key (camelCase format)

    Args:
        category: Log category enum

    Returns:
        Config key string (e.g., "pages")
    """
    mapping: dict[LogCategory, str] = {
        LogCategory.POLLING: "polling",
        LogCategory.PAGES: "pages",
        LogCategory.HOOKS: "hooks",
        LogCategory.AUTH: "auth",
        LogCategory.HTTP: "http",
        LogCategory.API: "api",
        LogCategory.ERRORS: "errors",
        LogCategory.SECURITY: "security",
        LogCategory.POLYGON_SEARCH: "polygonSearch",
        LogCategory.DOCUSIGN: "docusign",
    }
    return mapping[category]


def is_always_enabled(category: LogCategory) -> bool:
    """
    Check if a category should always be enabled (errors and security)

    Args:
        category: Log category enum

    Returns:
        True if category is always enabled
    """
    return category in (LogCategory.ERRORS, LogCategory.SECURITY)
